### PR TITLE
I2 54 - SSG / SSR

### PR DESCRIPTION
**Pull request by Aleksandrlitvinov**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new `HomePage` component for a more structured home layout.
  - Added a `PublicPage` component to display public posts with server-side fetching.
  - Developed a `PublicPostDescription` component for managing post descriptions with toggle functionality.
  - Created a `TotalUsersCounter` component to display the total number of registered users.

- **Enhancements**
  - Improved localization structure for home page strings in English and Russian.
  
- **Bug Fixes**
  - Adjusted rendering conditions for various profile display components.

- **Styling**
  - Added new styles for user profile headers and public post descriptions, enhancing overall UI consistency.

- **Documentation**
  - Included Storybook stories for the `TotalUsersCounter` and related components to facilitate testing and documentation.

### CODE REVIEW

File: src/shared/ui/ProfilePosts/ProfilePosts.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  profilePosts: GetPostResponse[]
+}
+
+export const ProfilePosts = ({ isMyPost, isShowPostId, profilePosts }: PropsType) => {
+  return (
+    <div className={s.postsList}>
+      {profilePosts.map(post => (
+        <ViewPostModal
+          author={post.author}
+          comments={[]}
```
Comment (ZaikouskiSergey) [inline src/shared/ui/ProfilePosts/ProfilePosts.tsx]: в comments пустой массив всегда, а он же не обязательный, ты вроде объяснял, но глянь, если все ок то ок

File: src/shared/ui/PublicPostDescription/PublicPostDescription.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <div className={s.descriptionContainer}>
+      <Typography
+        as={'p'}
+        className={clsx(s.text, { [s.fullText]: isFullText })}
+        variant={'regularText14'}
+      >
+        {isFullText ? `${descriptionText}...` : `${shortText}...`}
+        {!isShowFullSize && (
+          <span className={s.toggleButton} onClick={toggleText}>
+            {isFullText ? 'Hide' : 'Show more'}
```
Comment (ZaikouskiSergey) [inline src/shared/ui/PublicPostDescription/PublicPostDescription.tsx]: нужно потом будет подправить не строки а перевод

**Review by ZaikouskiSergey — APPROVED** [review]

**Review by Smartestbox — APPROVED** [review]

File: src/shared/ui/ProfilePosts/ProfilePosts.tsx (hunk 1 above)
Comment (Aleksandrlitvinov) [reply src/shared/ui/ProfilePosts/ProfilePosts.tsx]: Исправил

File: src/shared/ui/PublicPostDescription/PublicPostDescription.tsx (hunk 2 above)
Comment (Aleksandrlitvinov) [reply src/shared/ui/PublicPostDescription/PublicPostDescription.tsx]: Добавил в locales

**Pull request merged**